src/base_alloc/base_alloc_linear.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#include <assert.h>
+#include <stdint.h>
+
+#include "base_alloc_internal.h"
+#include "base_alloc_linear.h"
+#include "utils_common.h"
+#include "utils_concurrency.h"
+
+// minimum size of a single pool of the linear base allocator
+#define MINIMUM_LINEAR_POOL_SIZE (ba_os_get_page_size())
+
+// alignment of the linear base allocator
+#define MEMORY_ALIGNMENT (8)
+
+// metadata of the linear base allocator
+typedef struct {
+    size_t pool_size;
+    os_mutex_t lock;
+    char *data_ptr;
+    size_t size_left;
```

Comment:
you probably don't need size left - you can calculate that based on pool size and data_ptr. not sure whether that's worth changing.

Reply:
I prefer to have it saved, than calculate it each time - it is more clear for me.